doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+Testing the boot chain
+**********************
+
+To test the bootloader(s) and the secure boot chain, build and program them with any sample as described above.
+
+By default, both |NSIB| and MCUboot print information to their serial output on boot.
+This output will include information about the validation of images in its slots, as well as firmware-specific information if using :option:`CONFIG_FW_INFO` with |NSIB|.
+To see this output:
+
+1. |connect_terminal|
+#. Reset the development kit.
+#. Observe that each bootloader in the chain prints its information upon boot (some values may vary by build):
+
+.. tabs::
+
+   .. tab:: |NSIB|
+
+      .. code-block::
+
+         Attempting to boot slot 0.
+         Attempting to boot from address 0x9000.
+         Verifying signature against key 0.
+         Hash: 0xc0...71
+         Firmware signature verified.
+         Firmware version 1
+         Setting monotonic counter (version: 1, slot: 0)
+
+   .. tab:: MCUboot
+
+      .. code-block::
+
+         [00:00:00.359,039] <inf> mcuboot: Starting bootloader
+         [00:00:00.365,295] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.375,671] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.386,169] <inf> mcuboot: Boot source: none
+         [00:00:00.391,815] <inf> mcuboot: Swap type: none
+         [00:00:00.420,166] <inf> mcuboot: Bootloader chainload address offset: 0xc000
+         [00:00:00.428,039] <inf> mcuboot: Jumping to the first image slot
+
+When compiled with minimal configurations that disable logging output, such as ``prj_minimal.conf``, the bootloader information output can be disabled altogether or per-bootloader.
+Refer to the source code directories of each bootloader to see what minimal configuration options are already available, or add them through a custom Kconfig fragment if desired.
+
+.. _ug_bootloader_adding_fragments:
+
+Bootloader configurations and fragments
+***************************************
+
+You can use overlays specific to bootloaders in combination with project configurations and Kconfig fragments using :ref:`ug_multi_image_variables`.
+
+For example, you can assign the ``my-custom-overlay.conf`` fragment to the |NSIB|, which uses the child name of ``b0``, as follows:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -Db0_OVERLAY_CONFIG=my-custom-overlay.conf
+
+In the same way, you can replace ``b0`` with ``mcuboot`` to apply the ``my-custom-overlay.conf`` fragment to the MCUboot image.
+This is applicable for any child image in the build, as well as any Kconfig option given from the command line.
+
+See :ref:`ug_multi_image_variables` for more information about customizing images further with this method.
+
+.. _ug_bootloader_adding_external_flash:
+
+External flash memory partitions
```

Comment:
The title of this section suggests a reference concept section, while in fact it is a task: Using external flash memory partitions. I strongly suggest rephrasing the title.

Reply:
Fixed